---
float32[4] l_intr_p
float32[4] r_intr_p
float32[9] r2l_r
float32[3] r2l_t
float32[5] l_k
float32[5] r_k
bool success
string message
